### ISSUE TITLE
Accelerate does not work with MaskGenerationPipelone

### ISSUE THREAD

**Issue by MrinalTyagi** [huggingface/transformers#25808]
### System Info

- `transformers` version: 4.29.2
- Platform: Linux-5.15.0-46-generic-x86_64-with-glibc2.17
- Python version: 3.8.17
- Huggingface_hub version: 0.15.1
- Safetensors version: not installed
- PyTorch version (GPU?): 2.0.1 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: yes
- Using distributed or parallel set-up in script?: yes

Using device_map with mask-generation pipeline of sam results in error. 

@Narsil 

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

Run example SAM pipeline available at https://huggingface.co/docs/transformers/main/model_doc/sam with device_map="auto"

### Expected behavior

It should run with device_map="auto" otherwise if not possible should provide a suitable error.

**Comment by ArthurZucker** [comment]
According to the contribution guideline, could you share a complete reproducer as well as a traceback of the error you are guetting?
